vue-client/src/components/mixins/sidesearch-mixin.vue

```diff
@@ -136,16 +127,49 @@ export default {
       this.fetch(n);
     },
     getItems(keyword) {
-      let params = this.getSearchParams(this.getSearchPhrase(keyword));
-      params = Object.assign(params, {
+      const searchPhrase = this.getSearchPhrase(keyword);
+      const urlSearchParams = new URLSearchParams({
+        q: searchPhrase,
         _limit: this.maxResults,
         _offset: this.currentPage * this.maxResults,
         _sort: this.sort,
+        ...(typeof this.typeArray !== 'undefined' && this.typeArray.length > 0 && { '@type': this.typeArray }),
+        ...this.currentSearchParam?.mappings,
+        ...this.currentSearchParam?.searchProps.reduce((acc, searchProp) => ({
+          ...acc,
+          [searchProp]: searchPhrase,
+        }), {}),
       });
-      if (typeof this.typeArray !== 'undefined' && this.typeArray.length > 0) {
-        params['@type'] = this.typeArray;
+
+      const field = VocabUtil.getTermObject(this.fieldKey, this.resources.vocab, this.resources.context);
+
+      /**
+       * If field is a kbv:predicate (e.g. role) then filter linkable items depending on field and parent type.
+       * */
+      if (field.subPropertyOf.find(subProp => subProp['@id'] === VocabUtil.getTermObject('predicate', this.resources.vocab, this.resources.context)['@id'])) {
+        const statement = VocabUtil.getTermObject(field.domain[0]['@id'], this.resources.vocab, this.resources.context); // e.g. Contribution
+        const statementOf = statement.allowedProperties.find(p => p.domain?.find(d => d['@id'] === statement['@id'])); // e.g. contributionOf
+        const statementOfRangeIncludes = statementOf.rangeIncludes[0]['@id']; // e.g. Endeavour
+        const subClassesOfRangeIncludes = VocabUtil.getSubClassChain(statementOfRangeIncludes, this.resources.vocabClasses, this.resources.context); 
+
+        const fieldParentPath = [...(this.path.split('.').slice(0, -2))].join('.'); // Maybe too specific... are we always sure we need to go up two steps to find field parent path?
```

Comment:
I'd say yes, given that we are within a reified relation, and that relation is directly on the "parent" entity we want here... (The path string repr which needs to be split and rebuilt is more awkward; but that's another level of redesign to address.)

Reply:
Made a small simplification in https://github.com/libris/lxlviewer/pull/893/commits/6291e0573bf0e55c82ef29d01c9b3c5f267f4b14 as strings already are immutable.

I have to think some more about a less awkward solution 🤔😊